scenario_player/runner.py

```diff
@@ -153,289 +335,326 @@ def local_seed(self) -> str:
+        token_network_address = maybe_create_token_network(
+            token_network_registry_proxy, token_proxy
         )
 
-        wait_for_txs(self.client, fund_tx | ud_token_tx | mint_tx)
+        log.info("Making sure all nodes have the same token network")
+        self.ensure_token_network_discovery(token_proxy, token_network_address)
 
-        if node_starter is not None:
-            log.debug("Waiting for nodes to finish starting")
-            node_starter.get(block=True)
+        log.info("Setup done, running scenario", token_network_address=token_network_address)
 
-        first_node = self.get_node_baseurl(0)
+        task_config = self.definition.scenario.root_config
+        task_class = self.definition.scenario.root_class
 
-        registered_tokens = set(
-            self.session.get(
-                API_URL_TOKENS.format(protocol=self.protocol, target_host=first_node)
-            ).json()
-        )
-        if self.token.checksum_address not in registered_tokens:
-            for _ in range(5):
-                code, msg = self.register_token(self.token.checksum_address, first_node)
-                if 199 < code < 300:
-                    break
-                # FIXME: Remove long sleep and instead intelligently handle Raiden 5 block wait
-                gevent.sleep(15)
-            else:
-                log.error("Couldn't register token with network", code=code, message=msg)
-                raise TokenRegistrationError(msg)
-
-        self.token_network_address = self.ensure_token_network_discovery()
-        log.info(
-            "Token Network Discovery Completed", token_network_address=self.token_network_address
-        )
-        log.info("Setup done")
-        # Start root task
-        self.root_task()
+        root_task = task_class(runner=self, config=task_config)
 
-    def run_scenario(self):
-        log.info("run scenario")
-        mint_gas = GAS_LIMIT_FOR_TOKEN_CONTRACT_CALL * 2
+        # Expose attributes used by the tasks
+        self.token = token_proxy
+        self.root_task = root_task
+        self.contract_manager = proxy_manager.contract_manager
+        self.token_network_address = to_checksum_address(token_network_address)
+        self.block_execution_started = block_execution_started
 
-        with Janitor() as nursery:
-            self.node_controller = NodeController(self, self.definition.nodes, nursery)
-            fund_tx, node_starter, node_addresses, node_count = self._initialize_nodes(nursery)
+        root_task()
 
-            setup = nursery.spawn_under_watch(
-                self._setup, mint_gas, node_count, node_addresses, fund_tx, node_starter
+    def setup_raiden_nodes_ether_balances(
+        self, pool: Pool, node_addresses: Set[ChecksumAddress]
+    ) -> Set[Greenlet]:
+        """ Makes sure every Raiden node has at least `NODE_ACCOUNT_BALANCE_MIN`. """
+
+        greenlets: Set[Greenlet] = set()
```

Comment:
Is the set necessary? 
Shouldn't the pool keep track of the added greenlets by itself?